__tests__/app.test.js

```diff
@@ -201,3 +201,72 @@ describe("PATCH /api/articles/:article_id", () => {
 	  });
 	})
   })
+
+describe("POST /api/articles/:article_id/comments", () => {
+	test("returns a correctly formatted comment object", () => {
+		const commentObject = {
+			username: "butter_bridge",
+			body: "Leeds United Champions League Champions 2024"
+		}
+		return request(app)
+		.post("/api/articles/2/comments")
+		.send(commentObject)
+		.expect(201)
+		.then(({body}) => {
+			const {comment} = body
+			expect(comment).toEqual(
+				expect.objectContaining({
+					comment_id: expect.any(Number),
+					body: expect.any(String),
+					article_id: expect.any(Number),
+					author: expect.any(String),
+					votes: expect.any(Number),
+					created_at: expect.any(String),
+
+				})
+			)
+		})
+	})
+	test("returns with 404 when article_id not found", () => {
+		const commentObject = {
+			username: "butter_bridge",
+			body: "Leeds United Champions League Champions 2024"
+		}
+		return request(app)
+		.post("/api/articles/9999999/comments")
+		.send(commentObject)
+		.expect(404)
+		.then(({ body }) => {
+			const { msg } = body;
+			expect(msg).toEqual("Not found");
+		});
+	})
+	test("returns with 404 when username not found", () => {
+		const commentObject = {
+			username: "Dave_Hockaday",
+			body: "Leeds United Champions League Champions 2024"
+		}
+		return request(app)
+		.post("/api/articles/2/comments")
+		.send(commentObject)
+		.expect(404)
+		.then(({ body }) => {
+			const { msg } = body;
+			expect(msg).toEqual("Not found");
+		});
+	})
+	test("returns 400 when passed invalid article_id format", () => {
+		const commentObject = {
+			username: "Dave_Hockaday",
+			body: "Leeds United Champions League Champions 2024"
+		}
+		return request(app)
+		.post("/api/articles/MassimoCellino/comments")
+		.send(commentObject)
+		.expect(400)
+		.then(({ body }) => {
+			const { msg } = body;
+			expect(msg).toEqual("Bad request");
+		});
+	})
```

Comment:
all we need now is a test for the case where any of the required fields are missing